bucket/filezilla.json

```diff
@@ -15,9 +15,31 @@
     },
     "extract_dir": "FileZilla-3.47.1",
     "pre_install": [
-        "$file = 'fzdefaults.xml'",
-        "if (-not (Test-Path \"$persist_dir\\$file\")) {",
-        "    Copy-Item \"$dir\\docs\\fzdefaults.xml.example\" \"$dir\\fzdefaults.xml\"",
+        "@'",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\" ?>",
+        "<FileZilla3>",
+        "  <Settings>",
+        "    <Setting name=\"Config Location\">config</Setting>",
+        "    <Setting name=\"Disable update check\">1</Setting>",
+        "  </Settings>",
+        "</FileZilla3>",
+        "'@ | Out-File \"$dir\\fzdefaults.xml\" -Encoding Ascii",
+        "",
+        "$oldfzdefault = \"$persist_dir\\fzdefaults.xml\"",
+        "if (Test-Path \"$oldfzdefault\") {",
+        "    Write-Host 'Fixing existing configuration...' -f Yellow",
+        "    $oldConfigDir = \"$((Get-Item \"$persist_dir\").PSDrive.Root)filezilla\"",
+        "    $newConfigDir = \"$persist_dir\\config\"",
+        "    if (Test-Path \"$oldConfigDir\") {",
+        "        Write-Host \"Moving existing configuration from '$oldConfigDir' to '$newConfigDir'\" -f Yellow",
+        "        Move-Item \"$oldConfigDir\" \"$newConfigDir\"",
+        "    }",
+        "    if (Compare-Object $(Get-Content \"$oldfzdefault\") $(Get-Content \"$dir\\docs\\fzdefaults.xml.example\")) {",
```

Comment:
- Why do you need `compare-object`?
	- Only to make sure that user did not changed it?
- Subshells are not needed

Reply:
Yes just to be sure to not delete a file that would have been manually edited by the user. I will look to remove the subshells.

Reply:
Then do not delete the file in any use case.

Just rename it all the time